libs/wingsdk/src/shared-aws/api.ts

```diff
@@ -78,20 +79,19 @@ export class ApiEndpointHandler {
   /**
    * Returns a `cloud.Function` handler for handling requests from a `cloud.Api`.
    * @param handler The `onRequest` handler.
-   * @param corsOptions The CORS options.
+   * @param headers HTTP response headers to add to all responses (used by CORS)
    * @returns The `cloud.Function` handler.
    */
   public static toFunctionHandler(
     handler: cloud.IApiEndpointHandler,
-    corsOptions?: cloud.ApiCorsOptions
+    headers?: Record<string, string>
   ): cloud.IFunctionHandler {
-    return convertBetweenHandlers(
-      handler,
-      join(__dirname, "api.onrequest.inflight.js"),
-      "ApiOnRequestHandlerClient",
-      {
-        corsHeaders: cloud.Api.renderCorsHeaders(corsOptions)?.defaultResponse,
-      }
-    );
+    return lift({ handler }).inflight(async (ctx, request) => {
+      return apigwFunctionHandler(
+        request as unknown as APIGatewayProxyEvent,
+        ctx.handler,
+        headers
+      ) as unknown as string;
+    });
```

Comment:
`apigwFunctionHandler` is from an upper scope so this will break at runtime.

It may be possible to solve this by getting the path to that util (need to import `path` as well)

```suggestion
    return lift({ handler, utilPath: join(__dirname, "api-util") }).inflight(async (ctx, request) => {
      return require(ctx.utilPath).apigwFunctionHandler(
        request as unknown as APIGatewayProxyEvent,
        ctx.handler,
        headers
      ) as unknown as string;
    });
```

Reply:
OK!